client-api/src/main/java/edu/harvard/hms/dbmi/avillach/hpds/data/query/Query.java

```diff
@@ -28,15 +28,88 @@ public Query(Query query) {
 		this.id = query.id;
 	}
 
-	public ResultType expectedResultType = ResultType.COUNT;
-	public List<String> crossCountFields = new ArrayList<String>();
-	public List<String> fields = new ArrayList<String>();
-	public List<String> requiredFields;
-	public List<String> anyRecordOf;
-	public Map<String, DoubleFilter> numericFilters;
-	public Map<String, String[]> categoryFilters;
-	public List<VariantInfoFilter> variantInfoFilters;
-	public String id;
+	private ResultType expectedResultType = ResultType.COUNT;
+	private List<String> crossCountFields = new ArrayList<>();
+	private List<String> fields = new ArrayList<>();
+	private List<String> requiredFields = new ArrayList<>();
+	private List<String> anyRecordOf = new ArrayList<>();
+	private Map<String, DoubleFilter> numericFilters = new HashMap<>();
+	private Map<String, String[]> categoryFilters = new HashMap<>();
+	private List<VariantInfoFilter> variantInfoFilters = new ArrayList<>();
+	private String id;
+
+
+	public ResultType getExpectedResultType() {
+		return expectedResultType;
+	}
+
+	public List<String> getCrossCountFields() {
+		return crossCountFields;
+	}
```

Comment:
Should we do something like:
```suggestion
	public List<String> getCrossCountFields() {
	        if (crossCountFields == null) {
	                crossCountFields = new ArrayList<>();
	        }
		return crossCountFields;
	}
```
to avoid null pointers?

Reply:
I added checks to the getters to make sure the field is never null. It made sense to me to do it there where we are making defensive copies, since we are modifying collections in the query right now (sorting), which isn't ideal.